tests/testTriggers.go

```diff
@@ -4341,6 +4345,176 @@ func createLongevityJiraIssue(event *EventRecord, err error) {
+						if err != nil {
+							err = fmt.Errorf("error getting kvdb members using node %v, cause: %v ", nodes[0].Name, err)
+							logrus.Error(err.Error())
+
+						}
+						m, ok := newKvdbMembers[id]
+
+						if !ok && len(newKvdbMembers) > 0 {
+							logrus.Infof("node %v is no longer kvdb member", kvdbNode.Name)
+							isKvdbStatusUpdated = true
+
+						}
+						if ok && !m.IsHealthy {
+							logrus.Infof("kvdb node %v isHealthy?: %v", id, m.IsHealthy)
+							isKvdbStatusUpdated = true
+						} else {
+							logrus.Infof("Waiting for kvdb node %v health status to update to false after PX is stopped", kvdbNode.Name)
+							time.Sleep(1 * time.Minute)
+						}
+
+						waitTime--
+
+					}
+					if err != nil {
+						logrus.Error(err)
+						UpdateOutcome(event, err)
+
+					}
+
+					waitTime = 15
+					isKvdbMembersUpdated := false
+
+					for !isKvdbMembersUpdated && waitTime > 0 {
+						newKvdbMembers, err := Inst().V.GetKvdbMembers(nodes[0])
+						if err != nil {
+							err = fmt.Errorf("error getting kvdb members using node %v, cause: %v ", nodes[0].Name, err)
+							logrus.Error(err.Error())
+						}
+
+						_, ok := newKvdbMembers[id]
+
+						if ok {
+							logrus.Infof("Node %v still exist as a KVDB member. Waiting for failover to happen", kvdbNode.Name)
+							time.Sleep(2 * time.Minute)
+						} else {
+							logrus.Infof("node %v is no longer kvdb member", kvdbNode.Name)
+							isKvdbMembersUpdated = true
+
+						}
+
+						waitTime--
+					}
+
+					if err != nil {
+						logrus.Error(err)
+						UpdateOutcome(event, err)
+					}
+
+					newKvdbMembers, err := Inst().V.GetKvdbMembers(nodes[0])
+					if err != nil {
+						logrus.Error(err)
+						UpdateOutcome(event, err)
+
+					}
+					kvdbMemberStatus := validateKVDBMembers(event, newKvdbMembers, false)
+					errorChan = make(chan error, errorChannelSize)
+
+					if !kvdbMemberStatus {
+						logrus.Infof("Skipping remaining Kvdb node failovers as not all members are healthy")
+						StartVolDriverAndWait([]node.Node{kvdbNode}, &errorChan)
+						for err = range errorChan {
+							UpdateOutcome(event, err)
+						}
+						break
+					}
+
+					StartVolDriverAndWait([]node.Node{kvdbNode}, &errorChan)
+					for err = range errorChan {
+						logrus.Infof("Error a")
+						UpdateOutcome(event, err)
+					}
+
+				}
+			} else {
+				err = fmt.Errorf("not all kvdb members are healthy")
+				logrus.Errorf(err.Error())
+				UpdateOutcome(event, err)
+			}
+
+		})
+	})
+}
+
+func validateKVDBMembers(event *EventRecord, kvdbMembers map[string]*volume.MetadataNode, isDestuctive bool) bool {
+	logrus.Infof("Current KVDB members: %v", kvdbMembers)
+
+	allHealthy := true
+
+	if len(kvdbMembers) == 0 {
+		return false
+	}
+
+	for id, m := range kvdbMembers {
+		logrus.Infof("validating node : %v, is leader?: %v", id, m.Leader)
+
+		if !m.IsHealthy {
+			err := fmt.Errorf("kvdb member node: %v is not healthy", id)
+			allHealthy = allHealthy && false
```

Comment:
this is the same as `allHealthy = false` ?

Reply:
no.. here even if one node is not healthy, allhealthy will be false, if we set Healthy = false, there is a chance that second node is not healthy and third is healthy , then allHealthy will be true which is not the actual case